### PR TITLE
Added Azure DevOps CI/CD definition

### PR DESCRIPTION
**Pull request by vsafonkin**
This PR add pipeline file for Azure DevOps.
Azure pipeline contents steps:
- select xcode version from matrix for build
- get info about ruby and bundler
- install dependences and build app
- exec cucumber tests
- publish results

Travis and Jenkins files was deleted.
Also device list for cucumber tests was changed: now it has iphone simulators for latest models.
Calabash framework was updated from 0.21.7 to 0.21.8

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  - script: |
+      set -e
+      echo -e "install: --no-document --env-shebang\nupdate:  --no-document --env-shebang" > ~/.gemrc
+      bundle install
+    displayName: "bundle install"
+    
+  - script: cd ./CalSmokeApp && bundle install
+    displayName: "CalSmokeApp bundle install"
+
+  - script: cd ./CalSmokeApp && make app-cal
```
Comment (maxim-lobanov) [inline azure-pipelines.yml]: instead of `cd` use `workingDirectory` property

Comment (maxim-lobanov) [reply azure-pipelines.yml]: - script: 'bundle install'
  workingDirectory: 'CalSmokeApp'
  displayName: CalSmokeApp bundle install

File: CalSmokeApp/Gemfile [hunk 2]
```diff
@@ -8,7 +8,7 @@ gem "json", "1.8.6"
 # workflow and build tools
 gem 'retriable', '~> 2.0'
 gem 'rake', '~> 10.3'
-gem 'bundler', '~> 1.6'
+gem 'bundler', '~> 2.0'
```
Comment (jmoody) [inline CalSmokeApp/Gemfile]: bundler 1.17.3

Comment (vsafonkin) [reply CalSmokeApp/Gemfile]: This bundler version make a mistake in ADO: https://msmobilecenter.visualstudio.com/Mobile-Center/_build/results?buildId=782779

Comment (vsafonkin) [reply CalSmokeApp/Gemfile]: Done!

**Pull request merged**